projects/plugins/boost/wp-js-data-sync.php

```diff
@@ -128,7 +128,7 @@ function jetpack_boost_initialize_datasync() {
  * Register Data Sync Stores
  */
 jetpack_boost_register_option( 'critical_css_state', $critical_css_state_schema );
-jetpack_boost_register_option( 'critical_css_suggest_regenerate', Schema::as_boolean()->fallback( false ) );
+jetpack_boost_register_option( 'critical_css_suggest_regenerate', Schema::as_string()->fallback( '' ) );
```

Comment:
What do you think of `Schema::as_enum( array( ... allowed values ) )->nullable()`? That way `null` is considered as no notice needed, then there is a set of allowed values. I believe `nullable` also auto-falls back to `null` if an invalid value is supplied.

Then you can make the typescript type `z.coerce.enum( [ ... allowed types ... ] ).nullable()` too. :)

Reply:
Is d5cf219602 ok?

Reply:
Thank you for this! I Was unable to do it, probably because I was trying to coerce an enum and didn't think to include `nullable`. Your solution is much simpler and worked as expected during my tests.